src/config/deployments.ts

```diff
@@ -40,31 +40,31 @@ export type EthrDidRegistryDeployment = {
  * Represents the known deployments of the ERC1056 registry contract.
  */
 export const deployments: EthrDidRegistryDeployment[] = [
-  { chainId: 1, registry: '0xdca7ef03e98e0dc2b855be647c39abe984fcf21b', name: 'mainnet', legacyNonce: true },
-  { chainId: 3, registry: '0xdca7ef03e98e0dc2b855be647c39abe984fcf21b', name: 'ropsten', legacyNonce: true },
-  { chainId: 4, registry: '0xdca7ef03e98e0dc2b855be647c39abe984fcf21b', name: 'rinkeby', legacyNonce: true },
-  { chainId: 5, registry: '0xdca7ef03e98e0dc2b855be647c39abe984fcf21b', name: 'goerli', legacyNonce: true },
-  { chainId: 42, registry: '0xdca7ef03e98e0dc2b855be647c39abe984fcf21b', name: 'kovan', legacyNonce: true },
-  { chainId: 30, registry: '0xdca7ef03e98e0dc2b855be647c39abe984fcf21b', name: 'rsk', legacyNonce: true },
-  { chainId: 31, registry: '0xdca7ef03e98e0dc2b855be647c39abe984fcf21b', name: 'rsk:testnet', legacyNonce: true },
+  { chainId: BigInt(1), registry: '0xdca7ef03e98e0dc2b855be647c39abe984fcf21b', name: 'mainnet', legacyNonce: true },
```

Comment:
Hmm... I definitely get the irritation around the bigint usage for chainIDs and I also don't 100% understand the decision to use it for that specific thing, but I'd not like to deviate from ethers there. It's their domain, which they own and there might be a good reason for it. Especially due to the behaviour of losing precision when reading very long integers from JSON into a javascript number that might lead to very intransparent problems. If the whole ecosystem moves with ethers you'd always need to translate it.

Thinking about this, one would also need to implement such logic (like reading the deployment configuration from a file) so why not solve this on that level then?

Reply:
For the cases where the chainId would be too big to fit into a number, we still have the option of specifying it as a hex string. Do you have an example of such a network?

Since this is a config for ethr-did-resolver, it makes sense to have some logic for parsing it in this library instead of deferring to ethers.

As an example where such a config gets loaded from a file, we have the Veramo CLI.
In that case, the agent configuration is specified as a YML file which dynamically loads all the plugins specified in that file including their configs. That includes the configuration for this resolver where there is no opportunity to run any extra code on that level.

Reply:
I pushed the alternative number configuration! please review :)